src/nvim/ex_cmds.c

```diff
@@ -5296,9 +5297,11 @@ void ex_helptags(exarg_T *eap)
   }
 
   /* Get a list of all files in the help directory and in subdirectories. */
-  STRCPY(NameBuff, dirname);
+  if (STRLCPY(NameBuff, dirname, MAXPATHL) >= MAXPATHL)
+    EMSG(_(e_pathtoolong));
   add_pathsep((char *)NameBuff);
-  STRCAT(NameBuff, "**");
+  if (STRLCAT(NameBuff, "**", MAXPATHL) >= MAXPATHL)
+    EMSG(_(e_pathtoolong));
```

Comment:
Here, I believe the code below will already print an error that it couldn't find any matching files. I think this area is more descriptive, but perhaps we should control the printing below via a boolean.